classes/middleware/SetupContextBasedOnRequestUrl.php

```diff
@@ -29,12 +29,9 @@ class SetupContextBasedOnRequestUrl
      *
      * @example
      *      1. admin section (failed job list) : index.php/index/api/v1/jobs/all
-     *      2. admin section (create context) : index.php/_/api/v1/contexts
```

Comment:
Are you sure the `_` in URLs is appropriately removed/changed? I don't recall this part in detail, but I do recall there being special consideration for `_` in the URL distinct from the context path. It's probably not a good idea to make a breaking change in API URLs without a lot of thought/consideration first!

Reply:
Yep, this is another thing that I'm going to review before merging (I'll do self-review and write some final comments later). I found it curious to have both :)

Reply:
@asmecher I didn't flag anything special regarding the `_`, there are such checks in place: `($contextPath === 'index' || !$contextPath || $contextPath === Application::CONTEXT_ID_ALL)`

The origin of the `_` is here (previously `*` was being used, then it caused problems and got replaced by `_`): https://github.com/pkp/pkp-lib/issues/4414#issuecomment-764669427

The only drawback is that it's currently possible to use `index` as a journal path (there are 31 journals using such name on the Beacon).

Reply:
I think it's interesting to reserve the word `index` and force these journals to change, the name conflicts with the site-wide frontend (in a multi-journal installation, an `index` journal will be shadowed) and might pose other unexpected situations (regarding those checks that I've shared above).

If you agree, I'll leave the code this way and create an extra issue to complete the handling (pre-flight migration to force users to change the path).

Reply:
I agree that the use of `_` is an outlier and if we were introducing it afresh I'd prefer to stick with `index` for consistency -- but this does apply to the public API and would break existing integrations that use the `contexts` endpoint and probably others. See the REST API documentation: https://docs.pkp.sfu.ca/dev/api/ojs/3.4#tag/Access

Reply:
Hmm, given the documentation doesn't talk much about it, I think few people might be using this route 🤔

So....:
- I'll keep the updates for the `index`
- Keep this route functional (it's already this way, `_` might be replaced by `index`), but deprecated.
- Create another issue to make the `index` a reserved path.

Reply:
I've created this issue https://github.com/pkp/pkp-lib/issues/10241 and I'll push a PR to update the documentation.